### PR TITLE
creating project and adding a new project process is working

### LINKED ISSUE #56: 28 - Create user-project-backend
[MEMBER PROJECT](https://www.figma.com/file/MpfNES0KuBUkkd0Qk4EyOc/HYF-Connect-Wireframe?node-id=276%3A219)

### PR DESCRIPTION
**Pull request by Moamin-AbuEwaida**
the backend side for creating a project, calling all projects is working perfectly, just remaining calling a project with projetcID

### CODE REVIEW

File: api/controllers/project.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+res.send(JSON.stringify(allProjects, null, 2));
+} catch (error) {
+res.status(500).send(error);
+}
+},
+getProjectById: async (req, res) => {
+try {
+const projectId = req.params.projectId;
+
+const project = await projectManager.getProject(projectId);
```
Comment (dirk-jacobs) [inline api/controllers/project.js]: The function is called getProjectById

File: api/business-logic/project.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+createProject: async ({ Title, Description,GithubURL,WebsiteURL,Thumbnail}) => {
+const newProject = { Title, Description,GithubURL,WebsiteURL,Thumbnail };
+await ProjectStore.create(newProject);
+return newProject;
+},
+getAllProjects: async () => {
+const allProjects = await ProjectStore.findAll();
+return allProjects;
+},
+getProjectbyId: async(projectId) =>{
```
Comment (dirk-jacobs) [inline api/business-logic/project.js]: Use correct  camel-case:
getProjectById iso getProjectbyId

File: api/business-logic/project.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+const project = await ProjectStore.find(projectId);
```
Comment (dirk-jacobs) [inline api/business-logic/project.js]: Use findByPk in this case

File: api/access.log [hunk 4]
```diff
@@ -292,3 +292,20 @@
 ::ffff:127.0.0.1 - - [22/Dec/2021:18:59:42 +0000] "GET /pages/homepage/banner-section.js HTTP/1.1" 200 548 "http://127.0.0.1:3000/pages/homepage/homepage.js" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/96.0.4664.110 Safari/537.36"
```
Comment (dirk-jacobs) [inline api/access.log]: PLease delete this file and add it to .gitignore

File: api/routes/project.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
+const express = require("express");
+const projectController = require("../controllers/project");
+
+const projectRoutes = express.Router();
+
+projectRoutes.get("/", projectController.get);
+projectRoutes.get("/:projectId", projectController.getProjectById)
+projectRoutes.post("/addProject", projectController.post);
```
Comment (dirk-jacobs) [inline api/routes/project.js]: The default for a post is an insert
For me, "addProject" is not needed

This is ok
projectRoutes.post("/", projectController.post);

**Review by dirk-jacobs — CHANGES_REQUESTED** [review]
Next to a number of changes, there is a problem with the autoincrement in case you inserted a project and specified a ProjectID because appearantly the 'internal counter' is not incremented in this case. Solutions found so far:
- start with an empty db
- Retry a number of times because with every try, the internal counter is incremented. After about 5 times, it's starting to work
- Don't specify the ProjectID during creation of the initial records and hope they will be 1,2,3

https://github.com/sequelize/sequelize/issues/5954

**Comment by Moamin-AbuEwaida** [comment]
Thanks @dirk-jacobs for the followup comments
I edited the branch files accordingly but I still have the same thing for calling a certain project by projectId at postman and I receive the array of all the project :thinking: 
another weird response I have is when I'm checking the DB using pgAdmin tool, I still can't see any project added to the table even the 1st 3 project added during the testing of scripts>>create-project.js

File: api/routes/index.js [hunk 6]
```diff
@@ -1,18 +1,21 @@
-"use strict";
-
 const express = require("express");
 const router = express.Router();
+<<<<<<< HEAD
```
Comment (dirk-jacobs) [inline api/routes/index.js]: There seems to be a merge conflict

**Review by dirk-jacobs — CHANGES_REQUESTED** [review]
There seems to be a merge confoct in index.js which needs to be fixed first

File: api/routes/index.js (hunk 6 above)
Comment (Moamin-AbuEwaida) [reply api/routes/index.js]: I'll check it out

File: api/business-logic/project.js [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+    GithubURL,
+    WebsiteURL,
+    Thumbnail,
+  }) => {
+    const newProject = { Title, Description, GithubURL, WebsiteURL, Thumbnail };
+    await ProjectStore.create(newProject);
+    return newProject;
+  },
+  getProjectById: async (projectId) => {
+    try {
```
Comment (rayanejsilva) [inline api/business-logic/project.js]: You can remove try/catch. You already moved this part to controllers.

**Review by rayanejsilva — APPROVED** [review]
LGTM!

**Pull request merged**